README.md

```diff
@@ -73,18 +73,241 @@ Here are the results of our trained model.
+image augmentation and label conversion tasks we have created a [`data_preparer.py`](utils/data_preparer.py).
+With this script, we easily augmented the [4Tzones Traffic Lights Dataset](https://yadi.sk/d/a1Kr8Wmg0zfa0A),
+which dramatically increased the models' ability to generalize. 
+More about the [`data_preparer.py`](utils/data_preparer.py) script is in 
+the [Data Preparer Script](#data-preparer-script) section. The resulting annotated augmented dataset,
+which is called "4Tzones Traffic Lights Augmented Dataset" can be downloaded using the link below and 
+is licensed under the 
+[Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)](https://creativecommons.org/licenses/by-sa/4.0/) 
+license.   
+
+|      \* & \*\*         |  4Tzones Traffic Lights Augmented Dataset  |  
+|------------------------|          :------------------------:        |  
+| Link                   | https://yadi.sk/d/q2Yyy9PO2SrMKQ           |  
+| License                | [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/) |  
+| Total TL # of Samples  | 11988                                      |  
+| Red TL # of Samples    | 1998                                       |  
+| Yellow TL # of Samples | 1998                                       |  
+| Green TL # of Samples  | 1998                                       |  
+| No TL # of Samples     | 5994                                       |  
+
+\* Notice that the number of samples of red, yellow, and green traffic lights is equal to each other 
+and the number of samples of images without traffic lights is triple that. 
+Such a balancing of the dataset is suggested in the
+[How to improve object detection](https://github.com/AlexeyAB/darknet#how-to-improve-object-detection)
+section of README file from [https://github.com/AlexeyAB/darknet](https://github.com/AlexeyAB/darknet) 
+repository by [AlexeyAB](https://github.com/AlexeyAB).  
+
+\** Notice that you cannot obtain the same level of "balancing" by only using 
+[`data_preparer.py`](utils/data_preparer.py) on the
+[4Tzones Traffic Lights Dataset](https://yadi.sk/d/a1Kr8Wmg0zfa0A). The process of creating the 
+[4Tzones Traffic Lights Augmented Dataset](https://yadi.sk/d/q2Yyy9PO2SrMKQ) involved several manual steps, 
+that can be described as follows.
+Initially we had stored labeled images from different ROS bags in different folders. On each dataset, 
+we performed flipping, scaling, and balancing. For balancing the number of samples per class was set to about 2000 
+(the [`data_preparer.py`](utils/data_preparer.py) script has a `--balance N` option which commands the script to
+create N sample images per each class though augmentation).
+Then, the script randomly picked 1998 / 3 = 666 (5994 / 3 = 1998 for images without traffic lights) samples. 
+After all, we combined augmented images defived from all three ROS bags into the 
```

Comment:
derived

Reply:
Reworked this sentence so that it does not contain the "derive" verb now.